api/client.go

```diff
@@ -422,9 +428,45 @@ func NewClient(c *Config) (*Client, error) {
 		client.token = token
 	}
 
+	// get and poll token from agent sink if it is available
+	if agentSinkPath := os.Getenv(EnvAgentFileSinkPath); agentSinkPath != "" {
```

Comment:
We haven't yet finalized the order of token sources, but if we choose to make VAULT_TOKEN the highest priority, we should put this in an else{} or something.